libsigner/src/session.rs

```diff
@@ -96,9 +96,9 @@ impl StackerDBSession {
     where
         F: FnOnce(&mut StackerDBSession, &mut TcpStream) -> R,
     {
-        if self.sock.is_none() {
-            self.connect_or_reconnect()?;
-        }
+        // TODO: fix this so we can use persistent connection
```

Comment:
I strongly suspect that the node is closing the socket after it dumps the event, but not on a predictable schedule (see https://github.com/stacks-network/stacks-blockchain/blob/master/testnet/stacks-node/src/event_dispatcher.rs#L126).  The socket gets consumed by `async_h1`, which presumably drops it and closes it at some future date.  A next-step on this issue would be to update the node to either (1) always close the socket _explicitly_ once the payload has been sent, or (2) require `async_h1` to keep it open so the TCP window can at least be reused.  I don't care which one happens, but it would at least let us address this comment.

Reply:
I have created a ticket already for this https://github.com/stacks-network/stacks-blockchain/issues/3922 . I will link to this comment for additional context :D